File: examples/tutorials/hello_user/usage/swift/hello_user_usage.swift
import XCTest
import Wrapper

func runHelloUserTests() {

    let user = UserInfo(userName: "John", userAge: 22)
    let young_user = UserInfo(userName: "Kate", userAge: 18)

    let host = Wrapper.Host()

    assert(host.hello(user: user) == "Hello John")
    assert(host.hello(user: young_user) == "Hi Kate")

    assert(host.welcome(user: user) == "Welcome John! Let me know if you want something.")
    assert(host.welcome(user: young_user) == "Welcome Kate! Let me know if you want something.")

    user.want_a_drink = true
    young_user.want_a_drink = true

    assert(host.welcome(user: user) == "Welcome John! Do you want cap of beer?")
    assert(host.welcome(user: young_user) == "Welcome Kate! Do you want cap of juice?")

}

#if os(Linux)
runHelloUserTests()
#elseif os(OSX)
class HelloUserTests: XCTestCase {
    func testRun() throws {
        runHelloUserTests()
    }
}
#endif

File: examples/tutorials/hello_user/usage/python/hello_user_usage.py
from example_lib.hello.hello_user import UserInfo, Host

user = UserInfo(user_name="John", user_age=22)
young_user = UserInfo(user_name="Kate", user_age=18)

host = Host()

assert host.hello(user=user) == "Hello John"
assert host.hello(user=young_user) == "Hi Kate"

assert host.welcome(user=user) == "Welcome John! Let me know if you want something."
assert host.welcome(user=young_user) == "Welcome Kate! Let me know if you want something."

user.want_a_drink = True
young_user.want_a_drink = True

assert host.welcome(user=user) == "Welcome John! Do you want cap of beer?"
assert host.welcome(user=young_user) == "Welcome Kate! Do you want cap of juice?"
